### PR TITLE
feat: create initial VS Code extension

### PR DESCRIPTION
**Pull request by nhedger**
This PR sets up an draft version for ARA's VS Code highlighting extension.

I'll come up with a contribution guide at a later stage, for now the instructions in the readme should be pretty straightforward.

Here's a short list of features:

- [x] Automated testing on GitHub Actions on all three major operating systems.
- [x] Automated release management with release-please (using conventional commits).
- [x] Builds a desktop extension and web extension.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,25 +1,110 @@
...
-      "configuration": "./language-configuration.json"
-    }],
-    "grammars": [{
-      "language": "ara",
-      "scopeName": "source.ara",
-      "path": "./syntaxes/ara.json"
-    }]
-  }
+    "name": "ara.vscode",
+    "displayName": "ARA",
```
Comment (azjezz) [inline package.json]:
```suggestion
    "displayName": "Ara Highlighting",
```

Comment (nhedger) [reply package.json]: My reasoning behing this was that the extension would eventually do more than highlighting.

Comment (azjezz) [reply package.json]: no, this would just be for highlighting really, LSP extension would be it's own.

Comment (azjezz) [reply package.json]: later we can create an extension pack that contains both.

Comment (nhedger) [reply package.json]: Fair enough.

File: package.json [hunk 2]
```diff
@@ -15,10 +15,17 @@
...
+        {
+            "name": "Nicolas Hedger",
+            "email": "nicolas@hedger.ch",
+            "url": "https://github.com/nhedger"
+        },
+        {
+            "name": "Saif Eddin Gmati",
+            "url": "https://github.com/azjezz"
+        }
+    ],
```
Comment (azjezz) [inline package.json]:
```suggestion
    "contributors": [
        {
            "name": "Nicolas Hedger",
            "email": "nicolas@hedger.ch",
            "url": "https://github.com/nhedger"
        },
        {
            "name": "Saif Eddin Gmati",
            "email": "azjezz@protonmail.com",
            "url": "https://github.com/azjezz"
        }
    ],
```

please include a link to all contributors on the repo as well like we do in the readme.

**Pull request merged**